lib/suggestion-list.coffee

```diff
@@ -73,23 +75,47 @@ class SuggestionList
   isActive: ->
     @active
 
-  show: (editor) =>
-    return if @active
+  show: (editor, options) =>
+    if atom.config.get('autocomplete-plus.suggestionListFollows') is 'Cursor'
+      @showAtCursorPosition(editor, options)
+    else
+      @showAtBeginningOfPrefix(editor, options)
+
+  showAtBeginningOfPrefix: (editor, {prefix}) =>
     return unless editor?
-    @destroyOverlay()
 
-    if atom.config.get('autocomplete-plus.suggestionListFollows') is 'Cursor'
-      marker = editor.getLastCursor()?.getMarker()
-      return unless marker?
+    bufferPosition = editor.getCursorBufferPosition()
+    bufferPosition = bufferPosition.translate([0, -prefix.length]) if @wordPrefixRegex.test(prefix)
+
+    if @active
+      unless bufferPosition.isEqual(@displayBufferPosition)
+        @displayBufferPosition = bufferPosition
+        @suggestionMarker?.setBufferRange([bufferPosition, bufferPosition])
     else
-      cursor = editor.getLastCursor()
-      return unless cursor?
-      position = cursor.getBeginningOfCurrentWordBufferPosition()
-      marker = @suggestionMarker = editor.markBufferPosition(position)
-
-    @overlayDecoration = editor.decorateMarker(marker, {type: 'overlay', item: this})
-    @addKeyboardInteraction()
-    @active = true
+      @destroyOverlay()
+      @displayBufferPosition = bufferPosition
+      marker = @suggestionMarker = editor.markBufferPosition(bufferPosition)
+
+      # HACK: When the marker is at the cursor position, it will move with the
```

Comment:
@nathansobo This is probably marker abuse. Let me know if there is a better way to handle this situation.